dotcom-rendering/src/server/lib/get-content-from-url.js

```diff
@@ -52,27 +46,32 @@ exports.default = getContentFromURL;
 
 /**
  * @param {string} requestUrl
- * @returns {string}
+ * @returns {URL | undefined} the parsed URL, or false if there’s no fully qualified path
  */
 const parseURL = (requestUrl) => {
 	const url = decodeURIComponent(requestUrl.split('/').slice(2).join('/'));
 
-	return requestUrl.startsWith('/AMP') ? url.replace('www', 'amp') : url;
+	try {
+		return new URL(
+			requestUrl.startsWith('/AMP') ? url.replace('www', 'amp') : url,
+		);
+	} catch (error) {
+		return undefined;
+	}
 };
 
 exports.parseURL = parseURL;
 
 /** @type {import('webpack-dev-server').ExpressRequestHandler} */
 exports.getContentFromURLMiddleware = async (req, res, next) => {
-	if (req.path.split('/').length > 2) {
-		const sourceURL = parseURL(req.originalUrl);
+	const sourceURL = parseURL(req.originalUrl);
+
+	if (!sourceURL) return next();
```

Comment:
This reinstate a previous check with ensured that the `url` query param was set. Here we rely on the fact that an invalid URL will be parsed to `undefined`.